flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -108,17 +101,31 @@ public String getClusterDescription() {
 
 	@Override
 	public ClusterClientProvider<String> retrieve(String clusterId) {
-		return createClusterClientProvider(clusterId);
+		final ClusterClientProvider<String> clusterClientProvider = createClusterClientProvider(clusterId);
+
+		try (ClusterClient<String> clusterClient = clusterClientProvider.getClusterClient()) {
+			LOG.info(
+				"Retrieve flink cluster {} successfully, JobManager Web Interface: {}",
+				clusterId,
+				clusterClient.getWebInterfaceURL());
+		}
+		return clusterClientProvider;
 	}
 
 	@Override
 	public ClusterClientProvider<String> deploySessionCluster(ClusterSpecification clusterSpecification) throws ClusterDeploymentException {
-		final ClusterClientProvider<String> clusterClient = deployClusterInternal(
+		final ClusterClientProvider<String> clusterClientProvider = deployClusterInternal(
 			KubernetesSessionClusterEntrypoint.class.getName(),
 			clusterSpecification,
 			false);
 
-		return clusterClient;
+		try (ClusterClient<String> clusterClient = clusterClientProvider.getClusterClient()) {
```

Comment:
I am not sure what is the benefits to migrate the log or are there any considerations to do so?

Reply:
If we print the log in `createClusterClientProvider`. Then when we call clusterClientProvider.getClusterClient() the logs will show. We create a new cluster, the log will show twice and confusing. So i want to separate the log of creating and retrieving.

The log is changed when `ClusterClientProvider` is introduced.

Reply:
I got it, then make this reason as the commit descriptions.